sky/serve/service.py

```diff
@@ -236,13 +239,10 @@ def _get_host():
         serve_state.set_service_status_and_active_versions(
             service_name, serve_state.ServiceStatus.SHUTTING_DOWN)
     finally:
-        process_to_kill: List[multiprocessing.Process] = []
-        if load_balancer_process is not None:
-            process_to_kill.append(load_balancer_process)
-        if controller_process is not None:
-            process_to_kill.append(controller_process)
         # Kill load balancer process first since it will raise errors if failed
         # to connect to the controller. Then the controller process.
+        process_to_kill = filter(None,
```

Comment:
```suggestion
        process_to_kill = filter(lambda proc: proc is not None,
```
for better readability

Reply:
Actually, after digging deep into the function, setting the first argument to `None` will not only filter out `None` but all false variables, which is not the suggested way to check `None` (always use `if xx is (not) None` to check this, reference [here](https://google.github.io/styleguide/pyguide.html#2144-decision)). Can we try sth like `process_to_kill=[proc for proc in [lbproc, controllerproc] if proc is not None]`?

<img width="567" alt="image" src="https://github.com/user-attachments/assets/c345c7c3-3eeb-4c61-9fdf-16b7a2462fe2">

Reply:
Good idea!